src/pipx/util.py

```diff
@@ -161,6 +161,7 @@ def run_subprocess(
     log_cmd_str: Optional[str] = None,
     log_stdout: bool = True,
     log_stderr: bool = True,
+    run_dir: Optional[str] = None,
```

Comment:
Using `Optional` is considered confusing. There's even linters for making it `Union[$thing, None]`, which better represents what it means. With modern Pythons, it's more convenient to do this:
```suggestion
    run_dir: str | None = None,
```
Though, for supporting older versions, adding `from __future__ import annotations` is needed.